tests/integration/components/registries/finalize-registration-modal/component-test.ts

```diff
@@ -0,0 +1,109 @@
+        assert.equal(registrationModel.createDoi, true);
+        // Click embargo radio button
+        await click('[data-test-embargo-button]');
+        // Check that the submit button is disabled again
+        assert.dom('[data-test-submit-registration-button]').isDisabled();
+
+        // Close the dialog
+        this.set('isOpen', false);
+    });
+
+    test('embargo registration', async function(assert) {
+        this.store = this.owner.lookup('service:store');
+        const registration = server.create('registration');
+
+        const registrationModel = await this.store.findRecord('registration', registration.id);
+        this.set('model', registrationModel);
+        this.set('isOpen', false);
+        await render(hbs`
+            <Registries::FinalizeRegistrationModal::Manager
+                @registration={{this.model}}
+                @isOpen={{this.isOpen}}
+                @renderInPlace={{true}}
+                as |manager|
+            >
+                <Registries::FinalizeRegistrationModal @manager={{manager}} />
+            </Registries::FinalizeRegistrationModal::Manager>
+        `);
+        // Open the dialog
+        this.set('isOpen', true);
+        await settled();
+
+        // Check that the submit button is disabled
+        assert.dom('[data-test-submit-registration-button]').isDisabled();
+        // Click embargo radio button
+        await click('[data-test-embargo-button]');
+        // Check that the submit button is disabled
+        assert.dom('[data-test-submit-registration-button]').isDisabled();
+        // Check that `embargoEndDate` of the registration model is null
+        assert.equal(registrationModel.embargoEndDate, null);
+        // Enter an invalid date for embargo
+        await fillIn('[data-test-embargo-date-input] > div > input', moment().format('MM/DD/YYYY'));
+        // Check that`embargoEndDate` of the registration model is null
+        // And that the submit button is disabled
+        assert.equal(registrationModel.embargoEndDate, null);
+        assert.dom('[data-test-submit-registration-button]').isDisabled();
+        // TODO: Uncomment these lines once the potential race condition for validated-input/date is resolved
+        // // Enter a valid date for embargo
+        // const validDate = moment().add('days', 15).startOf('day');
+        // await fillIn('[data-test-embargo-date-input] > div > input', validDate.format('MM/DD/YYYY'));
+        // await blur('[data-test-embargo-date-input] > div > input');
+        // assert.deepEqual(registrationModel.embargoEndDate, validDate.toDate());
+        // // Check that the submit button is enabled
+        // assert.dom('[data-test-submit-registration-button]').isNotDisabled();
+        // // Click immediate radio button
+        // await click('[data-test-immediate-button]');
+        // // Check that the submit button is enabled
+        // assert.dom('[data-test-submit-registration-button]').isNotDisabled();
```

Comment:
These lines are commented due to a potential race condition in tests:
When I run the tests locally in the browser without hiding the container, the test works fine. However, if `hide container` is selected, the test fails. Don't know how to fix  the problem yet.

Reply:
I've also run into problems like this witch trying to simulate keypress.